### PR TITLE
[RM C09] LCM Automation  - final update

### LINKED ISSUE #2816: [RM Ch 09] Cloud Infra LCM introduction
Introduce main elements and foundational guidelines for LCM for cloud infrastructure in telco environment

### PR DESCRIPTION
**Pull request by walterkozlowski**
Reflecting changes discussed in RM meetings and in the Wiki page https://wiki.anuket.io/display/HOME/Infrastructure+LCM+Automation
No further updates to this section is envisaged for Moselle unless requested by the reviews

Fixes #2816, #2848, #2849, #2850, #2087

### CODE REVIEW

**Review by ulikleber — CHANGES_REQUESTED** [review]
We should linewrap to 120 chars from the beginning.

**Comment by karinesevilla** [comment]
> We should linewrap to 120 chars from the beginning.

+1

File: doc/ref_model/chapters/chapter09.rst [hunk 1]
```diff
@@ -135,15 +135,56 @@ Automation
...
 
 The goals of LCM are to provide a reliable administration of a system from its provisioning, through its operational stage, to its final retirement. Key functions of Infrastructure LCM:
  -  Hybrid, Multi-Cloud support, that is, LCM works across physical, virtual, and cloud environments, supporting on-premise, cloud, and distributed environments (like Edge)
- -  Complete system life cycle control (Build, Monitor, Maintain, Retire)
+ -  Complete system life cycle control (Plan/Design, Build, Provision, Operate/Manage, Retire, Recycle/Scrap)
  -  Enablement for automation of most system maintenance tasks
 
+Key benefits of the Infrastructure LCM Automation are:
+ -  Agility: stadardisation of the LCM process by writing and running IaaC allows to quickly and easily develop, stage, and produce environments
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter09.rst:151]: @walterkozlowski typo ->standardisation

Comment (walterkozlowski) [reply doc/ref_model/chapters/chapter09.rst:151]: Standardization or standardisation - are both correct... (British English uses "standardisation")

Comment (ulikleber) [reply doc/ref_model/chapters/chapter09.rst:151]: .. but there should be an "n" in either of them.

Comment (karinesevilla) [reply doc/ref_model/chapters/chapter09.rst:151]: yes, the issue is about the 'n'

**Review by karinesevilla — APPROVED** [review]

**Review by pgoyal01 — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@ulikleber , I have doe linewrapping to 120 characters, as requested.  Can you please approve if you can?
@CsatariGergely , you have raised PR#2944 to linewrap to 120 characters Chapter 9 of RM while I have heavily edited and appended subsection of this chapter in the current PR#2929 (already approved by the community  from the contents perspective, in RM meeting) and I have linewrapped it to 120 characters.  How can we combine these two PRs without loosing the new contents from PR#2929 and formatting from PR#2944?

File: doc/ref_model/chapters/chapter09.rst [hunk 2]
```diff
@@ -113,37 +126,116 @@ There are the following requirement types:
...
+Key benefits of the Infrastructure LCM Automation are:
+ -  Agility: standardisation of the LCM process by writing and running IaaC allows to quickly and easily develop, stage, and produce environments
+ -  Operational Consistency: automation of lifecycle  results in consistently maintaining desired state, reduces the possibility of errors and 
+    decreases the chances of incompatibility issues within the infrastructure
+ -  Human related Risks Mitigation: automation reduces risks related to human errors, rogue activities, and safeguards the institutional knowledge 
+    from leakage in case any employee leaves the organization
+ -  Higher Efficiency: achieved by minimizing human inaccuracies and eliminating the lack of knowledge about infrastructure installed base and its configuration, using
+ the CI/CD techniques adapted to infrastructure 
+ -  Cost/time Saving: engineers save up on time and cost which can be wisely invested in performing higher-value jobs; additional cost savings on cloud more optimal 
+ use of cloud resources using LCM Automation
```
Comment (ulikleber) [inline doc/ref_model/chapters/chapter09.rst:189]: Indentation need correction:
```suggestion
 -  Higher Efficiency: achieved by minimizing human inaccuracies and eliminating the lack of knowledge about infrastructure installed base and its configuration, using
    the CI/CD techniques adapted to infrastructure 
 -  Cost/time Saving: engineers save up on time and cost which can be wisely invested in performing higher-value jobs; additional cost savings on cloud more optimal 
    use of cloud resources using LCM Automation
```

File: doc/ref_model/chapters/chapter09.rst [hunk 3]
```diff
@@ -113,37 +126,116 @@ There are the following requirement types:
...
+perspective (the left hand side of the diagram), Site/Physical layer is entirely owned by the Infrastructure Owner, the virtualised infrastructure 
+layer (IaaS) is shared between the Infrastructure Owner and the Cloud Provider. Similarly,  the container orchestration layer (CaaS) is shared between 
+the Cloud Provider and the Cloud Consumer / Tenant.   These relationships can be illustrated by a situation, where a telecom operator owns the physical 
+infrastructure on which an external cloud provider runs the virtualisation software (hypervisor).  Sharing CaaS layer between the Cloud Provider and 
+the Cloud Consumer reflects the fact that the container management/orchestration software like Kubernetes is lifecycled by the Cloud Provider 
+(for instance when scaling out containers) but also by the Cloud Consumer because of the very close lifecycle relationship between an application and 
+a container in this model. For instance, destroying an application means also destroying related containers, Hence CaaS can be also considered as a part 
+of the Application Orchestration layer.
```
Comment (ulikleber) [inline doc/ref_model/chapters/chapter09.rst:214]: This seems a bit confusing or I don't understand the sentence. 
Application orchestration should be independent. Scaling out containers should be done by Kubernetes. Destroying an application shouldn't affect the CasS or IaaS platform.

**Comment by CsatariGergely** [comment]
> @ulikleber , I have doe linewrapping to 120 characters, as requested. Can you please approve if you can? @CsatariGergely , you have raised PR#2944 to linewrap to 120 characters Chapter 9 of RM while I have heavily edited and appended subsection of this chapter in the current PR#2929 (already approved by the community from the contents perspective, in RM meeting) and I have linewrapped it to 120 characters. How can we combine these two PRs without loosing the new contents from PR#2929 and formatting from PR#2944?

I can do the merging in this branch if that is okay for you @walterkozlowski .

File: doc/ref_model/chapters/chapter09.rst (hunk 3 above)
Comment (walterkozlowski) [reply doc/ref_model/chapters/chapter09.rst:214]: @ulikleber, when a Cloud Consumer destroys application, then Kubernetes would destroy containers related to this application.  This sentence simply explains that  the container as as service layer is (form operational perspective) shared between Cloud Consumer and Cloud Provider

**Comment by walterkozlowski** [comment]
@CsatariGergely, please merge into this branch your changes related to linewrap, but PLEASE making sure that the contents proposed in this PR is not affected.  Thanks.

**Comment by jbhartley** [comment]
> @jbhartley, can you review and approve if you can, please?

@walterkozlowski reviewing, as requested.

File: doc/ref_model/chapters/chapter09.rst (hunk 3 above)
Comment (jbhartley) [reply doc/ref_model/chapters/chapter09.rst:214]: @ulikleber + @walterkozlowski I think that what is deemed to "application orchestration" or "infrastructure automation" is solely a function of the role of the user/operator and not the tool. So (1) if Cloud Provider is exposing IaaS then responsibility for CaaS devolves to the Tenant and then the CaaS layer is part of "application" or (2) if Cloud Provider is exposing CaaS to tenant then orchestrating just the CNFs on top of this. The purpose is to illustrate these options, but agree that it it critical that what ever is exposed Cloud Provider should not be able to be "disrupted" by the Tenant, which is the "independence" that must be maintained.
NOTE: no change to content for this review.

**Review by jbhartley — APPROVED** [review]
@walterkozlowski I have reviewed the chapter: "9. Infrastructure Operations and Lifecycle Management" and the section: "9.5. Automation" and believe content is good, but maybe there needs to be more thought on structure of chapter, so it has higher outline sections before more detailed coverage items such as "Configuration and Lifecycle Management" which is part of what would typically be covered within the "operational management" loop of the overall Cloud Infrastructure Management process.

**Comment by CsatariGergely** [comment]
> > @ulikleber , I have doe linewrapping to 120 characters, as requested. Can you please approve if you can? @CsatariGergely , you have raised PR#2944 to linewrap to 120 characters Chapter 9 of RM while I have heavily edited and appended subsection of this chapter in the current PR#2929 (already approved by the community from the contents perspective, in RM meeting) and I have linewrapped it to 120 characters. How can we combine these two PRs without loosing the new contents from PR#2929 and formatting from PR#2944?
> 
> I can do the merging in this branch if that is okay for you @walterkozlowski .

Merged to the linewrapped master with [6c58475](https://github.com/cntt-n/CNTT/pull/2929/commits/6c58475d1d65aa9461fc537f456b056d6ece7f23)

File: doc/ref_model/chapters/chapter09.rst (hunk 3 above)
Comment (ulikleber) [reply doc/ref_model/chapters/chapter09.rst]: I still find the sentence confusing, but I will not object.

**Pull request merged**